rollouts/controllers/rollout_controller.go

```diff
@@ -680,13 +707,49 @@ func (r *RolloutReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	if err := r.store.Init(); err != nil {
 		return err
 	}
-	// TODO: watch cluster resources as well
+
+	var containerCluster unstructured.Unstructured
+	containerCluster.SetGroupVersionKind(configConnectorContainerClusterGVK)
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&gitopsv1alpha1.Rollout{}).
 		Owns(&gitopsv1alpha1.RemoteRootSync{}).
+		Watches(
+			&source.Kind{Type: &containerCluster},
+			handler.EnqueueRequestsFromMapFunc(r.mapClusterUpdateToRequest),
```

Comment:
You are an advance controller author now :)

Reply:
Working on it :)